backend/tracim_backend/app_models/validator.py

```diff
@@ -1,11 +1,93 @@
+import re
+import typing
+
+from marshmallow import ValidationError
+from marshmallow.validate import Length
+from marshmallow.validate import Regexp
 from marshmallow.validate import OneOf
-from tracim_backend.app_models.contents import content_type_list
+from marshmallow.validate import Range
+from marshmallow.validate import Validator
 
+from tracim_backend.app_models.contents import GlobalStatus
+from tracim_backend.app_models.contents import content_status_list
+from tracim_backend.app_models.contents import content_type_list
 # TODO - G.M - 2018-08-08 - [GlobalVar] Refactor Global var
 # of tracim_backend, Be careful all_content_types_validator is a global_var !
+from tracim_backend.exceptions import TracimValidationFailed
+from tracim_backend.models.auth import Profile
+from tracim_backend.models.auth import User
+from tracim_backend.models.data import ActionDescription
+from tracim_backend.models.data import UserRoleInWorkspace
 
-all_content_types_validator = OneOf(choices=[])
+class TracimValidator(object):
+
+    def __init__(self):
+        self.validators = {}
+        self.values = {}
+
+    def add_validator(self, field:str, value:str, validator:typing.Callable) -> bool:  # nopep8
+        if not value:
+            return False
+
+        self.validators[field] = validator
+        self.values[field] = value
+        return True
 
+    def validate_all(self):
+        errors = []
+        for field, value in self.values.items():
 
+            try:
+                self.validators[field](value)
+            except ValidationError as e:
+                errors.append(
+                    'Validation of {field} failed : {msg}'.format(
+                        field=field,
+                        msg=str(e)
+                    )
+                )
+        if errors:
+            raise TracimValidationFailed(str(errors))
+        return True
+
+
+
+
+## TODO - G.M - 2018-11-29 - Refactor validator system
+# static validator #
+# int
+bool_as_int_validator = Range(min=0, max=1, error="Value must be 0 or 1")
+strictly_positive_int_validator = Range(min=1, error="Value must be positive")
+positive_int_validator = Range(min=0, error="Value must be positive or 0")
+
+# string
+regex_string_as_list_of_int = Regexp(regex=(re.compile('^(\d+(,\d+)*)?$')))
```

Comment:
Can you add a comment about what do the regex ?

Reply:
done